lib/eventFactory.js

```diff
@@ -73,6 +73,42 @@ function getJobsFromPR(config) {
     return jobsToStart;
 }
 
+/**
+ * Starts the build if the changed file is part of the sourcePaths, or if there is no sourcePaths
+ * @method startBuild
+ * @param  {Object}   config                configuration object
+ * @param  {Object}   config.buildConfig    Build Config to create the build with
+ * @param  {Array}    config.changedFiles   List of files that were changed
+ * @param  {Array}    config.sourcePaths    List of soure paths
+ * @return {Promise}
+ */
+function startBuild(config) {
+    /* eslint-disable global-require */
+    const BuildFactory = require('./buildFactory');
+    const buildFactory = BuildFactory.getInstance();
+    /* eslint-enable global-require */
+    const { buildConfig, changedFiles, sourcePaths } = config;
+    let hasChangeInSourcePaths = true;
+
+    // Only check if sourcePaths exists
+    if (sourcePaths) {
+        hasChangeInSourcePaths = changedFiles.some(file =>
+            sourcePaths.some((source) => {
+                let path = source;
+
+                // make sure it's in this format foo/bar/
+                // because with changedFile as 'foo/barTest.md', it should return false
```

Comment:
Wondering if it's possible for a `sourcePath` to be an exact path to a file, because this logic would break in that case